src/open_inwoner/accounts/views/contactmoments.py

```diff
@@ -127,9 +137,20 @@ def get_anchors(self) -> list:
 
     def get_context_data(self, **kwargs):
         ctx = super().get_context_data(**kwargs)
-        kcm = fetch_klantcontactmoment_for_bsn(
-            kwargs["kcm_uuid"], self.request.user.bsn
-        )
+
+        if self.request.user.bsn:
+            kcm = fetch_klantcontactmoment_for_bsn(
+                kwargs["kcm_uuid"], self.request.user.bsn
+            )
+        elif self.request.user.kvk:
+            kvk_or_rsin = self.request.user.kvk
+            config = OpenKlantConfig.get_solo()
+            if config.use_rsin_for_innNnpId_query_parameter:
+                kvk_or_rsin = self.request.user.rsin
+            kcm = fetch_klantcontactmoment_for_kvk_or_rsin(
+                kwargs["kcm_uuid"], kvk_or_rsin
+            )
+
```

Comment:
- We're performing the same check in the list and detail view. 
- The code for `fetch_klantcontactmoment_for_bsn` and `fetch_clantcontactmoment_for_kvk_or_rsin` in `wrap.py` is (almost?) identical, ditto for the list versions.

Perhaps we could make the function more generic, i.e. `fetch_klantcontactmoment`, which works with `bsn`, `kvk`, and `rsin`. We could perform this check and grab the relevant data in the `KlantContactMomentBaseView` and store it in the context as `USER_IDENTIFIER`, which could be `bsn`, `kvk`, or `rsin`. Then we could do `kcm = fetch_klant_contactmoment(USER_IDENTIFIER)`.

Reply:
you're right, it was a bit too much copy pasting. I modified it https://github.com/maykinmedia/open-inwoner/pull/904/commits/904b902501c56f418201106a5a051aed3b00760f, I chose not to add the fetch parameters to the context data, because this check is also performed in another view (+ we don't want to expose things in `context` that we're not gonna use in our templates).